examples/http.rs

```diff
@@ -0,0 +1,116 @@
+use std::collections::HashMap;
+
+use anyhow::Result;
+use hyper::header::CONTENT_TYPE;
+use serde::{Deserialize, Serialize};
+
+use esp_idf_svc::hal::prelude::Peripherals;
+use esp_idf_svc::log::EspLogger;
+use esp_idf_svc::wifi::{BlockingWifi, EspWifi};
+use esp_idf_svc::{eventloop::EspSystemEventLoop, nvs::EspDefaultNvsPartition};
+
+use lynx_embedded::{reqwesp, wifi as espWifi};
+
+#[allow(dead_code)]
+#[derive(Deserialize, Debug)]
+struct Anything {
+    pub data: String,
+    pub json: Option<Account>,
+    pub method: String,
+    pub origin: String,
+    pub url: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Account {
+    username: String,
+    password: String,
+}
+
+fn main() -> Result<()> {
+    // Bind the log crate to the ESP Logging facilities
+    EspLogger::initialize_default();
+
+    // Configure Wifi
+    let peripherals = Peripherals::take().unwrap();
+    let sys_loop = EspSystemEventLoop::take()?;
+    let nvs = EspDefaultNvsPartition::take()?;
+
+    let mut wifi = BlockingWifi::wrap(
+        EspWifi::new(peripherals.modem, sys_loop.clone(), Some(nvs))?,
+        sys_loop,
+    )?;
+
+    espWifi::connect(&mut wifi)?;
```

Comment:
This is similar to what you did last PR right? Should we make a wrapper function. I understand this is just an example but is there a good justification why we duplicate code?

Reply:
This is an example target. It basically acts as an alternative `main.rs`, which is why it has some duplicate code as `main.rs`. In any binary, this block of code will only ever be called once.

Any files in the `examples` folder can be executed as if it is main by using the `--example` flag. E.g. `cargo run --example http` to run `http.rs` as main.

Reply:
Can we add this to the README when we update it?